var/spack/repos/builtin/packages/bufr/package.py

```diff
@@ -11,37 +13,108 @@ class Bufr(CMakePackage):
     utilities that can be used to read (decode) and write (encode)
     data in BUFR, which is a WMO standard format for the exchange of
     meteorological data. This is part of the NCEPLIBS project.
-
+    The library also includes a Python interface.
     """
 
     homepage = "https://noaa-emc.github.io/NCEPLIBS-bufr"
     url = "https://github.com/NOAA-EMC/NCEPLIBS-bufr/archive/refs/tags/bufr_v11.5.0.tar.gz"
+    git = "https://github.com/NOAA-EMC/NCEPLIBS-bufr"
 
-    maintainers("t-brown", "AlexanderRichert-NOAA", "edwardhartnett", "Hang-Lei-NOAA", "jbathegit")
+    maintainers("AlexanderRichert-NOAA", "edwardhartnett", "Hang-Lei-NOAA", "jbathegit")
 
+    version("develop", branch="develop")
+    version("12.0.1", sha256="525f26238dba6511a453fc71cecc05f59e4800a603de2abbbbfb8cbb5adf5708")
+    version("12.0.0", sha256="d01c02ea8e100e51fd150ff1c4a1192ca54538474acb1b7f7a36e8aeab76ee75")
     version("11.7.1", sha256="6533ce6eaa6b02c0cb5424cfbc086ab120ccebac3894980a4daafd4dfadd71f8")
     version("11.7.0", sha256="6a76ae8e7682bbc790321bf80c2f9417775c5b01a5c4f10763df92e01b20b9ca")
     version("11.6.0", sha256="af4c04e0b394aa9b5f411ec5c8055888619c724768b3094727e8bb7d3ea34a54")
     version("11.5.0", sha256="d154839e29ef1fe82e58cf20232e9f8a4f0610f0e8b6a394b7ca052e58f97f43")
+    version("11.4.0", sha256="946482405e675b99e8e0c221d137768f246076f5e9ba92eed6cae47fb68b7a26")
+
+    # Patch to not add "-c" to ranlib flags when using llvm-ranlib on Apple systems
+    patch("cmakelists-apple-llvm-ranlib.patch", when="@11.5.0:11.6.0")
+    # C test does not explicity link to -lm causing DSO error when building shared libs
+    patch("c-tests-libm.patch", when="@11.5.0:11.7.0")
+    # Patch to identify Python version correctly
+    patch("python-version.patch", when="+python")
+
+    variant("python", default=False, description="Enable Python interface?")
+    variant("shared", default=True, description="Build shared libraries", when="@11.5:")
+
+    extends("python", when="+python")
+
+    depends_on("python@3:", type=("build", "run"), when="+python")
+    depends_on("py-setuptools", type="build", when="+python")
+    depends_on("py-numpy", type="build", when="+python")
+    depends_on("py-pip", type="build", when="+python")
+    depends_on("py-wheel", type="build", when="+python")
+
+    def url_for_version(self, version):
+        pre = "bufr_" if version < Version("12.0.1") else ""
+        return (
+            f"https://github.com/NOAA-EMC/NCEPLIBS-bufr/archive/refs/tags/{pre}v{version}.tar.gz"
+        )
+
+    # Need to make the lines shorter at least on some systems
+    def patch(self):
+        with when("@:11.7.1"):
+            filter_file("_lenslmax 120", "_lenslmax 60", "CMakeLists.txt")
+
+    def cmake_args(self):
+        args = [
+            self.define_from_variant("ENABLE_PYTHON", "python"),
+            self.define_from_variant("BUILD_SHARED_LIBS", "shared"),
+            self.define("BUILD_TESTS", self.run_tests),
```

Comment:
So I understand, you only want BUILD_TESTS enabled when the installer runs `spack install --test=root`.  Is that correct?

Reply:
Correct, but just to check-- @climbfuji it looks like you added the "tests" variant, do you have any objection to doing it just through `spack install --test`? Currently it doesn't look like it actually installs the tests, it just builds them and then they'll get wiped out unless using `--keep-stage`.

Reply:
No objections at all, thanks for asking!

Reply:
That makes sense based on the nature of the check.  IF you have tests you want to cache from the build for use as stand-alone tests (days to weeks post-build), you'll want to check out the docs at https://spack.readthedocs.io/en/latest/packaging_guide.html#saving-build-time-files.